api/controllers/ReferralController.js

```diff
@@ -76,60 +78,31 @@ module.exports = {
    * platform. ie. sms,fb, mobile app
    */
   findCustomUrl: function(req, res) {
-    let customUrl = req.params.customUrl;
-    let resBody = { customUrl: req.params.customUrl, referralCount: 0 };
-    UserReferralCodesTwo.findOne({ code: customUrl })
-      .then(function(result) {
-        if (typeof result === "undefined") {
+    return Promise.coroutine(function*() {
+      let customUrl = req.params.customUrl;
+      let resBody = { customUrl: req.params.customUrl, referralCount: 0 };
+      let user = yield UserReferralCodesTwo.findOne({ code: customUrl });
+      try {
+        let smsUser;
+        let fbUser;
+        if (typeof user === "undefined") {
           throw new Error();
         }
-        // Find SMS subscribed users
-        if (result.platformSmsId) {
-          User.findOne({ id: result.platformSmsId })
-            .then(function(result) {
-              if (typeof result === "undefined") {
-                throw new Error();
-              }
-              Object.assign(resBody, result);
-              return User.count({
-                referredBy: result.phone
-              });
-            })
-            .then(function(referralCount) {
-              Object.assign(resBody, { referralCount: referralCount });
-              return res.json(resBody);
-            })
-            .catch(function(error) {
-              return res.json(404, {
-                error: "Unable to retrieve referral information for this user"
-              });
-            });
-        } else if (result.platformFbId) {
-          // Find facebook users
-          UserFb.findOne({ id: result.platformFbId })
-            .then(function(result) {
-              Object.assign(resBody, result);
-              return UserFb.count({ referredBy: customUrl });
-            })
-            .then(function(referralCount) {
-              Object.assign(resBody, { referralCount: referralCount });
-              return res.json(resBody);
-            })
-            .catch(function(error) {
-              return res.json(404, {
-                error:
-                  "Unable to retrieve information on the facebook user linked to this url"
-              });
-            });
+        if (user.platformSmsId) {
+          smsUser = yield User.findOne({ id: user.platformSmsId });
```

Comment:
I could be misunderstanding, but wouldn't we still need to do a `count` query to get the referral count? And maybe also a count on both the `referred_by` and `referred_by_v2` columns?